src/components/NoticeOfWithdrawal/ReferenceNumber.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div id="reference-number-section">
+    <v-row no-gutters>
+      <v-col
+        cols="12"
+        sm="3"
+        class="pr-4"
+      >
+        <label id="reference-number-label">
+          <strong>Folio or Reference Number</strong>
+        </label>
+      </v-col>
+      <v-col
+        cols="12"
+        :sm="9"
+      >
+        <v-form
+          id="reference-num-form"
+          ref="refNumRef"
+          v-model="valid"
+        >
+          <v-text-field
+            id="reference-number-input"
+            v-model="referenceNumber"
+            label="Folio or Reference Number (Optional)"
+            :rules="referenceNumRules"
+            filled
+          />
+        </v-form>
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Emit, Prop, Watch } from 'vue-property-decorator'
+import { FormIF } from '@bcrs-shared-components/interfaces'
+
+@Component({})
+export default class ReferenceNumber extends Vue {
+  // Refs
+  $refs!: Vue['$refs'] & {
+      refNumRef: FormIF
+  }
+
+  /** Prompt the validations. Used for global validations. */
+  @Prop({ default: false }) readonly autoValidation!: boolean
+
+  /** Draft court order number. */
+  @Prop({ default: '' }) readonly draftReferenceNumber!: string
+
+  // Local properties
+  private referenceNumber = ''
+  private referenceNumRules = []
+  private valid = false
+
+  /** Called when component is mounted. */
+  mounted (): void {
+    // Set default draft values if they exist
+    if (this.draftReferenceNumber) this.referenceNumber = this.draftReferenceNumber
+  }
+
+  /** Clear rules and reset validations. */
+  private clearValidations (): void {
+    this.referenceNumRules = []
+    this.$refs.refNumRef.resetValidation()
+  }
+
+  /** Triggers the form validation. */
+  public validate (): boolean {
+    return this.$refs.refNumRef.validate()
+  }
+
+  @Watch('autoValidation')
+  @Watch('referenceNumber')
+  validateReferenceNum (): void {
+    if (this.autoValidation) {
+      // Apply TextField rules
+      this.referenceNumRules = [
+        (v: string) => (!v || !/^\s/g.test(v)) || 'Invalid spaces', // leading spaces
+        (v: string) => (!v || !/\s$/g.test(v)) || 'Invalid spaces', // trailing spaces
+        (v: string) => (!v || !(v.length > 20)) || 'Folio or Reference number is invalid' // too long
+      ]
+      this.$refs.refNumRef.validate()
+    } else this.clearValidations()
+  }
+
+  /** Emit reference number. */
+  @Watch('referenceNumber')
+  @Emit('referenceNumber')
+  emitReferenceNumber (): void {}
```

Comment:
I know that other places follow this syntax/style for watchers/emitters but it looks wrong to me.

What you can't tell from this is the type of what's being emitted. The Watch decorator implicitly pushes a parameter into the function for the Emit decorator to emit.

I'd prefer to see this:
```
  @Watch('referenceNumber')
  @Emit('referenceNumber')
  emitReferenceNumber (referenceNumber: string): void {}
```

Does it work? What do you think?

Reply:
Or even
```
  @Watch('referenceNumber')
  @Emit('referenceNumber')
  emitReferenceNumber (val: string): void {}
```

Reply:
I thought about this.. but it shows an error the variable is declared but never used:
![image](https://github.com/user-attachments/assets/978f197d-db81-4b04-9d47-0d049a31b8a6)

What do you prefer?

Reply:
I prefer the verbose one. You can hide the error using eslint-ignore.

Reply:
Updated

Reply:
Update all of them?

Reply:
Sure, just updated